x-pack/platform/plugins/shared/stack_connectors/public/common/auth/auth_config.tsx

```diff
@@ -78,6 +84,41 @@ export const AuthConfig: FunctionComponent<Props> = ({ readOnly, isPfxEnabled =
 
   useEffect(() => setFieldValue('config.hasAuth', Boolean(authType)), [authType, setFieldValue]);
 
+  const options = [
+    {
+      value: null,
+      label: i18n.AUTHENTICATION_NONE,
+      'data-test-subj': 'authNone',
+    },
+    {
+      value: AuthType.Basic,
+      label: i18n.AUTHENTICATION_BASIC,
+      children: authType === AuthType.Basic && <BasicAuthFields readOnly={readOnly} />,
+      'data-test-subj': 'authBasic',
+    },
+    {
+      value: AuthType.SSL,
+      label: i18n.AUTHENTICATION_SSL,
+      children: authType === AuthType.SSL && (
+        <SSLCertFields
+          readOnly={readOnly}
+          certTypeDefaultValue={certTypeDefaultValue}
+          certType={certType}
+          isPfxEnabled={isPfxEnabled}
+        />
+      ),
+      'data-test-subj': 'authSSL',
+    },
+    isOAuth2Enabled && {
```

Comment:
Won't we need an intermediate release for this PR?

If so, I am thinking this could be

```suggestion
    (isOAuth2Enabled || authType === AuthType.OAuth2ClientCredentials) && {
```

1. `isOAuth2Enabled` should be false in an initial release
2. second release enables it
3. if there is a rollback to 1 the UI would be visible only when editing connectors previously configured with OAuth2

Reply:
I'm not sure I'm following. The "isOAuth2Enabled" prop is to show the OAuth2 select option field in the webhook connector, but not in the cases-webhook connector. It did not exist before, which is the same as all being false. Maybe you thought "isOAuth2Enabled" is a feature flag?

Reply:
> It did not exist before, which is the same as all being false.

Right, but in the case of changes that require an intermediate release, we need to deploy everything hidden first(or just the backend).

What I was proposing was:
- Release 1
  - All this PR, but we don't display anything yet in the UI.
  - What we **could** show is the form, if and only if, an existing connector has this configured
  - `(isOAuth2Enabled || authType === AuthType.OAuth2ClientCredentials)`
- Release 2
  - The UI is always shown.

So if release 2 is reverted after someone has already created a connector with OAuth2, the customer would still be able to edit their connector in release 1.